logstash-core/src/main/java/org/logstash/plugins/outputs/Stdout.java

```diff
@@ -57,7 +57,7 @@ public void output(final Collection<Event> events) {
                 do {
                     encodeCompleted = codec.encode(e, encodeBuffer);
                     outputStream.write(encodeBuffer.array(), encodeBuffer.position(), encodeBuffer.limit());
-                    encodeBuffer.flip();
+                    encodeBuffer.clear();
```

Comment:
was there a reason why flip was used before that doesn't make sense any more? I think I'm missing some context here about what is the issue being solved.

Reply:
It was just a mistaken use of the `Buffer` API. The `flip` method "flips" a buffer from write mode to read mode. In the line above, the buffer has already been read in the `outputStream.write` call so it needs to be flipped back from read mode to write mode. The `compact` and `clear` methods do this. Because the buffer is always read in its entirety, the more efficient `clear` method can be used here instead of `compact`.

This behavior can be easily verified in a unit test, so I added one for it.